styx-common/src/main/java/com/spotify/styx/util/BasicWorkflowValidator.java

```diff
@@ -66,15 +71,13 @@ public BasicWorkflowValidator(DockerImageValidator dockerImageValidator) {
     final List<String> e = new ArrayList<>();
 
     var componentId = workflowId.componentId();
-    if (componentId.isEmpty()) {
-      e.add("component id cannot be empty");
-    } else if (componentId.contains("#")) {
+
+    empty(e, "component id", workflowId.componentId());
```

Comment:
Should we rename this to something more descriptive? like checkIfEmpty or something

Reply:
or shouldNotBeEmpty

Reply:
I agreed that those are better names but I was being consistent with `limit`. We can make this change and limit in a following PR

Reply:
cool 👍